systemtests/src/main/java/com/github/streamshub/systemtests/utils/SetupUtils.java

```diff
@@ -0,0 +1,84 @@
+package com.github.streamshub.systemtests.utils;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.github.streamshub.systemtests.Environment;
+import com.github.streamshub.systemtests.logs.LogWrapper;
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.api.model.SecretBuilder;
+import io.skodjob.testframe.resources.KubeResourceManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+
+public class SetupUtils {
+    private static final Logger LOGGER = LogWrapper.getLogger(SetupUtils.class);
+    private SetupUtils() {}
+
+    // ---------------
+    // Yaml config
+    // --------------
+    public static String getYamlFileContent(String fileUrl) {
+        LOGGER.debug("Loading YAML file content from url {}", fileUrl);
+        StringBuilder content;
+        try (BufferedReader in = new BufferedReader(new InputStreamReader(new URL(fileUrl).openStream(), StandardCharsets.UTF_8))) {
+            content = new StringBuilder();
+            String inputLine;
+
+            while ((inputLine = in.readLine()) != null) {
+                content.append(inputLine).append("\n");
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot download YAML content from url: " + fileUrl, e);
+        }
+        return content.toString();
+    }
+
+    public static <T> T configFromYaml(File yamlFile, Class<T> c) {
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+        try {
+            return mapper.readValue(yamlFile, c);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    // --------------
+    // Namespace
+    // --------------
+    public static void copyImagePullSecrets(String targetNamespace) {
```

Comment:
Will there be any private images included in the testing?

Reply:
There is currently a public image in the config.yaml file. This only allowed usage of custom test client with specific kafka version. But i guess for our usage, we could go only the public image, so we can get rid of all of this around test client image being a custom one.